migrations/back-37/back-37.go

```diff
@@ -0,0 +1,207 @@
+	if err != nil {
+		logger.WithError(err).Error("Unable to execute distinct query")
+	} else {
+		logger.Debugf("Finding jellyfish records for %d users", len(userIDs))
+
+		for _, userID := range userIDs {
+			logger.Debugf("Finding jellyfish records for user ID %s", userID)
+			selector := bson.M{
+				"_userId": userID,
+				"_active": true,
+				// uploads aren't de-duped.
+				"type":          bson.M{"$ne": "upload"},
+				"_deduplicator": bson.M{"$exists": false},
+			}
+
+			var jellyfishResult bson.M
+			jellyfishDocCursor, err := m.dataRepository.Find(context.Background(), selector)
+			if err != nil {
+				logger.WithError(err).Error("Unable to find jellyfish results")
+			}
+			for jellyfishDocCursor.Next(context.Background()) {
+				err = jellyfishDocCursor.Decode(&jellyfishResult)
+				if err != nil {
+					logger.WithError(err).Error("Could not decode mongo doc")
+					errorCount++
+					continue
+				}
+
+				dupQuery := bson.M{
+					"_userId":       userID,
+					"_active":       true,
+					"_id":           jellyfishResult["_id"],
+					"_deduplicator": bson.M{"$exists": true},
+				}
+				dupCursor, err := m.dataRepository.Find(context.Background(), dupQuery)
+				if err != nil {
+					logger.WithError(err).Errorf("Could not query for duplicate datum %s.", jellyfishResult["_id"])
+					errorCount++
+					continue
+				}
+				if dupCursor.Next(context.Background()) {
+					if !m.DryRun() {
+						if err := m.archiveDocument(jellyfishResult["_id"]); err != nil {
+							logger.WithError(err).Error("Unable to archive jellyfish document")
+							errorCount++
+						}
+					}
+					archivedCount++
+				} else {
+					if !m.DryRun() {
+						if err := m.migrateDocument(jellyfishResult); err != nil {
+							logger.WithError(err).Error("Unable to migrate jellyfish document")
+							errorCount++
+						}
+					}
+					hashUpdatedCount++
+				}
+			}
+			if err := jellyfishDocCursor.Err(); err != nil {
+				logger.WithError(err).Error("error while fetching data. Please re-run to complete the migration.")
+				errorCount++
+			}
+			jellyfishDocCursor.Close(context.Background())
+		}
+	}
+
+	if err != nil {
+		logger.WithError(err).Error("Unable to migrate jellyfish documents")
+		errorCount++
+	}
+
+	return hashUpdatedCount, archivedCount, errorCount
+}
+
+func (m *Migration) migrateDocument(jfDatum bson.M) error {
+	var deduplicatorUpdate bson.M
+
+	switch jfDatum["type"] {
+	case "smbg", "bloodKetone", "cbg":
+		if len(fmt.Sprintf("%v", jfDatum["value"])) > 7 {
+			// NOTE: we need to ensure the same precision for the
+			// converted value as it is used to calculate the hash
+			val := jfDatum["value"].(float64)
+			mgdlVal := val*18.01559 + 0.5
+			mgdL := glucose.MgdL
+			jfDatum["value"] = glucose.NormalizeValueForUnits(&mgdlVal, &mgdL)
+
+			deduplicatorUpdate = bson.M{
+				"$set": bson.M{
+					"_deduplicator": bson.M{
+						"hash": CreateHash(jfDatum),
```

Comment:
Do we want to set the deduplicator name and version? Same below in `default` clause.